pkg/api/controllers/fileshare.go

```diff
@@ -549,6 +549,22 @@ func (f *FileShareSnapshotPortal) CreateFileShareSnapshot() {
 		return
 	}
 
+	// Check existence of fileshare snapshot name #931
+	filesnaps, err := db.C.ListFileShareSnapshots(ctx)
```

Comment:
I prefer you can move the logic into util/db.go, and add some test cases for checking with the code.

Reply:
Moved to util/db.go and added test cases